### DISCUSSION TITLE
HTTP Output with dynamic header

### DISCUSSION THREAD

**Discussion by sreisich** [fluent/fluent-bit#9486] [Q&A]
Hi,
is there a way to create such kind of output, where the <namespace_name> is the kubernetes namespace? I'm trying to find a solution for hours now...

```
[OUTPUT]
  Name http
  Match kube.*
  host parseable.parseable.svc.cluster.local
  http_User user
  http_Passwd password
  format json
  port 80
  header Content-Type application/json
  header X-P-TAG-tag1 DEV
  header X-P-Stream <namespace_name>
  uri /api/v1/ingest
  json_date_key timestamp
  json_date_format iso8601
```

I hope someone have an idea, how to solve this.
Thank you very much.

**Comment by patrick-stephens** [comment]
Did you try the record-accessor syntax?
https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/record-accessor

It depends if the http output supports it for that configuration option but that is the way to do it.

**Comment by sreisich** [comment]
I tried with:
`header X-P-Stream $kubernetes['namespace_name']`

But it seems that it's not supported. I get:
`Stream name validation failed: Stream name cannot contain special character: $`

So the `$kubernetes` is not interpreted as variable...

**Reply by patrick-stephens** [reply]
Yup so sounds like it does not support record accessor syntax for that specific option. There may already be a feature request for it if you have a look.
